downloader/artifact_downloader.go

```diff
@@ -147,16 +149,16 @@ func (ad *ConcurrentArtifactDownloader) download(jobs <-chan downloadJob, result
 
 // extractArchive extracts an archive using the tar CLI tool by piping the archive to the command's input.
 func (ad *ConcurrentArtifactDownloader) extractArchive(r io.Reader, targetDir string) error {
-	tarArgs := []string{
-		"-x",      // -x: extract files from an archive: https://www.gnu.org/software/tar/manual/html_node/extract.html#SEC25
-		"-f", "-", // -f "-": reads the archive from standard input: https://www.gnu.org/software/tar/manual/html_node/Device.html#SEC155
-	}
-	cmd := ad.CommandFactory.Create("tar", tarArgs, &command.Opts{
+	cmd := ad.CommandFactory.Create("unzip", []string{}, &command.Opts{
 		Stdin: r,
 		Dir:   targetDir,
 	})
 
 	if err := cmd.Run(); err != nil {
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			return fmt.Errorf("command failed with exit status %d (%s): %w", exitErr.ExitCode(), cmd.PrintableCommandArgs(), errors.New("check the command's output for details"))
+		}
 		return fmt.Errorf("%s failed: %s", cmd.PrintableCommandArgs(), err)
```

Comment:
> check the command's output for details

Actually, here the command won't print any output to the build log because `Stdout` and `Stderr` are nil on `cmd`.
+ we could use `%w` from warping the error if it isn't an `exec.ExitError`.
```suggestion
	if out, err := cmd.RunAndReturnTrimmedCombinedOutput(); err != nil {
		var exitErr *exec.ExitError
		if errors.As(err, &exitErr) {
			return fmt.Errorf("command failed with exit status %d (%s): %w", exitErr.ExitCode(), cmd.PrintableCommandArgs(), errors.New(out))
		}
		return fmt.Errorf("%s failed: %w", cmd.PrintableCommandArgs(), err)
	}
```

Reply:
a6e05c00863b4b4d060b2e80f43be7915b73ad9d